tools/toolchains/__init__.py

```diff
@@ -988,6 +988,8 @@ def compile_command(self, source, object, includes):
         _, ext = splitext(source)
         ext = ext.lower()
 
+        source = abspath(source) if PRINT_COMPILER_OUTPUT_AS_LINK else source
+
```

Comment:
Why is this change needed?

Reply:
Because the GCC gives out what he gets, i.e. when you are passing relative paths the gcc prints out relative paths and vice versa.